### ISSUE TITLE
Performance issue with NATS transporter (v2.x.x)

### ISSUE THREAD

**Issue by icebob** [moleculerjs/moleculer#1237]
### Discussed in https://github.com/moleculerjs/moleculer/discussions/1235

<div type='discussions-op-text'>

<sup>Originally posted by **mrprigun** August  8, 2023</sup>
Hello, I encountered certain obstacles in my use case while attempting to execute the molecular actions. Briefly, I have a service (gateway for the main app with `moleculer-web`)  that looks like this :

```js
module.exports = {
  name: "test",
  actions: {
    hello: {
      rest: "GET /hello",
      handler(ctx) {
        ctx.meta.$responseType = "text/plain";
        return "Hello Moleculer";
      }
    },

    second_call: {
      rest: "GET /second_call",
      handler(ctx) {
        // some actions
        return ctx.call('external.call', params_action_results_above);
      }
    },
  }
};
```

`external.call` is located on another node and invocation happens using NATS transporter. By design actions `hello` and `second_call` are gonna be loaded to process unique tasks. Before publishing it to production I made benchmarks for each action and received such results on my laptop:

- The `hello` action works pretty well, the result was ~600rps 
- But the second one `second_call` was very much degraded, I received just ~40rps

Also, I received similar results after deployment to a prod-like environment. I believe this occurs due to external triggering, which is anticipated, but why so much? I've tried to use different load-balancing strategies and `bulkhead` but didn't receive any significant improvements. Is there a possibility of configuring Moleculer to enhance this behavior or it's some kind of bug?

</div>

**Comment by icebob** [comment]
It looks that it's an issue with the `nats` library version 2.x.x. With the previous 1.x.x version the performance is fine.
There is an open issue (2 years ago) about it in NATS repo: https://github.com/nats-io/nats.js/issues/438

**Comment by mrprigun** [comment]
For more context, libs:
```
"moleculer": "0.14.31",
"nats": "2.15.1",
```

Used OS for local testing: MacOS 13.5
Service images in the k8s cluster are based on node:18-alpine

**Comment by mrprigun** [comment]
Already tried. Initially, it was using nats `1.4.12` and the results were pretty similar, that's why I moved to `2.15.1`.

**Comment by icebob** [comment]
What is the NATS server version?

**Comment by mrprigun** [comment]
`2.9.11`, if more concrete  `docker.io/bitnami/nats:2.9.11-debian-11-r0 `

**Comment by icebob** [comment]
plz try with the latest version 2.9.21

**Comment by icebob** [comment]
It's strange because I can reproduce this issue without any metrics, only with 2.x.x nats lib.

**Comment by icebob** [comment]
I've found the problem inside the `nats` library. The sending logic is changed to queue-based in 2.x.x version. Skipping this logic I could reach 30.000 msg/sec instead of 40 msg/sec. I've opened an issue in the NATS client repo: https://github.com/nats-io/nats.js/issues/581

**Comment by mrprigun** [comment]
This is a significant improvement, waiting for this fix then 😃 

Pushed my tests just in case https://github.com/mrprigun/moleculer-benchmark-test There are two dedicated nodes and nats in docker-compose. With enabled StatsD reported I receive ~`170rps`, with disabled reporter it's ~`2k rps` on my laptop. 

**Nats service:** 2.9.21
**Nats lib:** 2.15.1
**Node:** v18.15.0
**OS:** MacOS 13.5

**Comment by icebob** [comment]
NATS fixed the issue [in 2.16.0](https://github.com/nats-io/nats.js/releases/tag/v2.16.0), my results:

![image](https://github.com/moleculerjs/moleculer/assets/306521/6bbfefc4-9a74-4a99-9112-45bb1d65363f)
